### PR TITLE
Backend node

### PR DESCRIPTION
**Pull request by pankajsoni91**
It is not complete. Is in progress. PR is only being raised so that I can take input from you whenever required and you can give your valuable input whenever you have time.

### CODE REVIEW

File: src/client/routes/dashboard/index.tsx [hunk 1]
```diff
@@ -1,44 +1,42 @@
...
-  counter: number;
-}
+import { bindActionCreators } from 'redux';
+import { connect, Dispatch } from 'react-redux';
+import { actions } from './reducer';
+import WithLoader from 'components/WithLoader';
+import DashboardInfo from './components/DashboardInfo';
+const DashboardInfoWrapper = WithLoader(DashboardInfo);
 
 interface IProps {
```
Comment (kuldeepkeshwar) [inline src/client/routes/dashboard/index.tsx:10]: this should extend reduce state interface

File: src/client/routes/dashboard/index.tsx [hunk 2]
```diff
@@ -1,44 +1,42 @@
...
-  counter: number;
+  dashboard: any;
+  actions: any;
 }
 
-class Dashboard extends React.Component<IProps, IState> {
+class Dashboard extends React.Component<IProps, any> {
```
Comment (kuldeepkeshwar) [inline src/client/routes/dashboard/index.tsx:15]: avoid any

File: src/client/routes/dashboard/index.tsx [hunk 3]
```diff
@@ -1,44 +1,42 @@
...
-    };
   }
 
-  incrementCounter = () => {
-    this.setState((prevState: IState) => {
-      return {
-        counter : prevState.counter + 1,
-      };
-    });
+  componentWillMount() {
```
Comment (kuldeepkeshwar) [inline src/client/routes/dashboard/index.tsx]: use componentDidMount for fetch data

File: src/client/routes/dashboard/index.tsx [hunk 4]
```diff
@@ -1,44 +1,42 @@
...
-        <Button
-          onClick={this.incrementCounter}
-        >
-          +1
-        </Button>
-        <span>{counter}</span>
-      </Wrapper>
-    );
+    const { loading, data } = this.props.dashboard;
+    console.log('loading', loading);
```
Comment (kuldeepkeshwar) [inline src/client/routes/dashboard/index.tsx]: avoid console.log

File: src/client/routes/dashboard/index.tsx [hunk 5]
```diff
@@ -1,44 +1,42 @@
...
+    return <DashboardInfoWrapper loading={loading} data={data} />;
   }
 }
 
-export default Dashboard;
+const mapStateToProps = ({ dashboard }) => ({ dashboard });
+
+// TODO - need to check whether any need to be remove or not
+const mapDispatchToProps = (dispatch: Dispatch<any>) => ({
+  actions: {
```
Comment (kuldeepkeshwar) [inline src/client/routes/dashboard/index.tsx:34]: use simple action creator & avoid wrapping every action in actions object for simple use case

File: src/client/services/dashboard.ts [hunk 6]
```diff
@@ -0,0 +1,12 @@
+import axios from 'axios';
+
+export function getDashboardData(): Promise<any> {
+  return axios
+    .get('/v1/api')
+    .then(response => {
+      return response.data;
```
Comment (kuldeepkeshwar) [inline src/client/services/dashboard.ts]: return statement is not required here

File: src/server/models/error.ts [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+    message?: string,
+    status?: number,
+    code?: string,
+    authErrorCode?: string,
+    requestToken?: string
+  ) {
+    this.message = message;
+    this.status = status || 500;
+    this.code = code || '';
+    this.authErrorCode = authErrorCode;
```
Comment (kuldeepkeshwar) [inline src/server/models/error.ts]: authErrorCode ?

File: src/client/routes/dashboard/index.tsx (hunk 3 above)
Comment (pankajsoni91) [reply src/client/routes/dashboard/index.tsx]: Done

File: src/client/routes/dashboard/index.tsx (hunk 4 above)
Comment (pankajsoni91) [reply src/client/routes/dashboard/index.tsx]: Removed

File: src/server/models/error.ts (hunk 7 above)
Comment (pankajsoni91) [reply src/server/models/error.ts]: Not of any use. As we have error code.. if required some will add the same for project specific.

File: src/client/routes/dashboard/index.tsx (hunk 1 above)
Comment (pankajsoni91) [reply src/client/routes/dashboard/index.tsx:10]: Done.

File: src/server/models/error.ts (hunk 7 above)
Comment (kuldeepkeshwar) [reply src/server/models/error.ts]: please remove it as it is not required

**Pull request closed without merging**